BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua

```diff
@@ -0,0 +1,28 @@
+-- We need a way to check if the player gave up before the song properly ended.
+-- It doesn't look like the engine broadcasts any messages that would be helpful here,
+-- so we do the best we can by checking the song position.
+
+local af = Def.ActorFrame{
+	OffCommand=function(self)
+		local fail = (GAMESTATE:GetCurMusicSeconds() < GAMESTATE:GetCurrentSong():GetLastSecond())
+
+		if GAMESTATE:IsCourseMode() then
```

Comment:
Can we add comments on what this block is doing? Something like, "Always fail if we're not already on the last song. If we are on the last song, then we fall back to the condition above"

Reply:
sure, I added a comment!